doc/specs/#6049 - Font config objects.md

```diff
@@ -0,0 +1,98 @@
+
+In the initial stages of this feature, we will only allow two non-default font configurations: `italicFont` and `boldFont`.
+
+### Inheritance
+
+The font configuration class will be implemented such that it allows for inheritance between objects of the class. As described earlier, within the same profile, non-default font configuration objects will inherit from the default font configuration object.
+
+*Between* profiles, each profile's default font configuration object will inherit from the default font configuration object defined in `profiles.defaults`. As for the non-default font configuration objects, these are treated as a single setting. I.e. a profile's `italicFont` font configuration object does *not* inherit from the `italicFont` object defined in `profiles.defaults`, it is simply its own setting. The `italicFont` configuration object defined in `profiles.defaults` only applies to profiles that do not define their own `italicFont`.
+
+This design is in line with the way we handle `defaultAppearance` and `unfocusedAppearance`, and was deliberately chosen so we maintain consistency across these different classes.
+
+## UI/UX Design
+
+Users will be able to add font configurations to their profiles that will look something like this:
+
+```
+"font": // this is the default
+{
+    "fontFace": "Cascadia Mono",
+    "fontWeight": "Normal",
+    "fontSize": 12
+},
+"boldFont":
+{
+    "fontFace": "Consolas",
+    "fontWeight": "Extra-Bold"
+}
+"italicFont":
+{
+    "fontWeight": "Light"
+}
+```
+
+In the settings UI, the current way font settings are presented to the user can be preserved, and additional dropdowns can be added for `boldFont` and `italicFont`. Each dropdown will contain options to configure `fontFace` and `fontWeight`.
+
+## Further implementation details
+
+Currently, the `DxRenderer` is given a desired font and it creates a text layout for the default case and a separate text layout for the italic case. To support font configuration objects, the `DxRenderer` will need to be updated to be able to receive a collection of desired fonts, and it will create a separate text layout for each desired font. These text layouts will then be obtained by the renderer when needed, as it already does for the italic case.
+
+## Capabilities
+
+### Accessibility
+
+Should not affect accessibility.
+
+### Security
+
+Should not affect security.
+
+### Reliability
+
+We have had several issues when parsing `fontFace` from settings in the past. This will add further potential for parsing errors simply because there will be more places where we will need to parse `fontFace`. Unfortunate as it is, it is an unavoidable aspect of this feature.
+
+### Compatibility
+
+This feature changes the way we expect to parse font settings. However, we have to make sure we are still able to parse font settings the way they are currently, so as to not break functionality for legacy users.

```

Comment:
Do we have precedent for a settings-file-rewriter that pre-passes legacy configs and just rewrites them into the current format? (Instead of leaving multiple parse-mechanisms in the current format loader?)